api/src/main/java/com/netflix/iceberg/Tables.java

```diff
@@ -43,4 +43,13 @@ Table create(Schema schema,
                String tableIdentifier);
 
   Table load(String tableIdentifier);
+
+  /**
+   * Drop a table from iceberg catalog, the underlying implementation
+   * may or may not drop the corresponding metadata and data.
+   * @param tableIdentifier
+   */
+  void drop(String tableIdentifier);
```

Comment:
add a flag or properties map maybe to allow expressing options for dropping data/metadata so that the contract is clearer?